parsec/core/gui/greet_device_widget.py

```diff
@@ -174,9 +177,20 @@ def _on_button_send_email_clicked(self):
 
     def _on_send_email_success(self, job):
         # In theory the invitation address shouldn't have changed, but better safe than sorry
-        self.invite_addr = job.ret
-        self.button_send_email.setText(_("TEXT_GREET_DEVICE_EMAIL_SENT"))
-        self.button_send_email.setDisabled(True)
+
+        self.invite_addr, email_sent_status = job.ret
+
+        if email_sent_status == InvitationEmailSentStatus.SUCESS:
+            self.button_send_email.setText(_("TEXT_GREET_DEVICE_EMAIL_SENT"))
+            self.button_send_email.setDisabled(True)
+        else:
+            show_info(
```

Comment:
Would be nice to tell in `show_info` the error type (recipient or sender's fault)